packages/x-components/src/x-modules/search-box/components/placeholder-animations/fade.vue

```diff
@@ -0,0 +1,58 @@
+<template>
+  <transition
+    v-on="$listeners"
+    class="x-fade-placeholders"
+    name="x-fade-placeholders-"
+    mode="out-in"
+    appear
+  >
+    <slot />
+  </transition>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component } from 'vue-property-decorator';
+
+  /**
+   * Fades the element passed in the slot.
+   *
+   * @public
+   */
+  @Component
+  export default class Fade extends Vue {}
+</script>
+
+<style lang="scss" scoped>
+  $fade-duration: 0.3s;
+  $fade-leave-delay: 1s;
+
+  .x-fade-placeholders {
+    &--enter-active,
+    &--leave-active {
+      transition: all $fade-duration ease;
+    }
+
+    &--leave-active {
+      transition-delay: $fade-leave-delay;
+    }
+
+    &--enter {
+      opacity: 0;
+    }
+
+    &--leave-to {
+      opacity: 0;
+    }
+  }
+</style>
+
+<docs lang="mdx">
+## Examples
+
+The Fade component is intended to be used as a prop in the `SearchInputPlaceholder` component.
+
+```vue
+<SearchInputPlaceholder :animatedPlacholders="placeholders" :animation="Fade" />
```

Comment:
If you agree on moving these animations to the `src/components/animations`, this example should be changed and references to `placeholders` (like the class names) removed to make it more generic.